### PR TITLE
feat: Implement torch.column_stack

### LINKED ISSUE #22391: column_stack
- [ ] #9

### PR DESCRIPTION
**Pull request by Purity-E**
# PR Description 

This PR is for the torch.column_stack frontend function. I have added both the function and test.

## Related Issue 

Close #22391 

## Checklist 

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you follow the steps we provided?

### Socials: 

https://twitter.com/p_nyagweth

### CODE REVIEW

**Comment by jieunboy0516** [comment]
Hi @Purity-E , thanks a lot for your contribution to Ivy! would you mind fixing the `indexing_slicing_joining_mutating_ops.py` codes as I saw some code duplication/ unnecessary changes. Thanks.

File: ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py [hunk 1]
```diff
@@ -1,368 +1,373 @@
...
+                    num_or_size_splits=tuple(
+                        [chunk_size + remainder] + [chunk_size] * (chunks - 1)
+                    ),
+                    axis=dim,
+                )
+
+
+@to_ivy_arrays_and_back
+def column_stack(tensors, *, out=None):
+    return ivy.stack(tensors, axis=1, out=out)
```
Comment (jieunboy0516) [inline ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py:50]: according to the documentation:

```Creates a new tensor by horizontally stacking the tensors in tensors.

Equivalent to torch.hstack(tensors), except each zero or one dimensional tensor t in tensors is first reshaped into a (t.numel(), 1) column before being stacked horizontally.
```
could you please implement the `each zero or one dimensional tensor t in tensors is first reshaped into a (t.numel(), 1) column before being stacked horizontally` part please?

File: ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py [hunk 2]
```diff
@@ -1,1604 +1,1630 @@
...
+            other=xs[1],
+            backend_to_test=backend_fw,
+        )
+        
+# column stack
+@handle_frontend_test(
+    fn_tree="torch.column_stack",
+    xs_n_input_dtypes_n_unique_idx=_arrays_idx_n_dtypes(),
+)
+def test_torch_columnstack(
```
Comment (jieunboy0516) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py:1611]: The unit test code looks fine to me

File: ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py (hunk 1 above)
Comment (Purity-E) [reply ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py:50]: Okay.

**Comment by jieunboy0516** [comment]
Hi @Purity-E , could you please improve the `column_stack` implementation and remove the `array_api` changes please? (as you can see from the `https://github.com/unifyai/ivy/pull/22714/files`) Thanks

File: ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py (hunk 1 above)
Comment (jieunboy0516) [reply ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py:50]: thanks a lot @Purity-E

**Comment by Purity-E** [comment]
> Hi @Purity-E , could you please improve the `column_stack` implementation and remove the `array_api` changes please? (as you can see from the `https://github.com/unifyai/ivy/pull/22714/files`) Thanks

I'm a bit confused with changes in the array_api. It was modified when I synced my forked repo with the upstream

**Comment by jieunboy0516** [comment]
@Purity-E maybe you can download the whole ivy repo again, make your column_stack changes and force push to your branch (even its not a good practice)

**Comment by Purity-E** [comment]
> @Purity-E maybe you can download the whole ivy repo again, make your column_stack changes and force push to your branch (even its not a good practice)

What if undo the commits using git revert or git reset? Is this recommendable?

**Comment by jieunboy0516** [comment]
Hi @Purity-E, thanks for the fix. The code looks good to me now! However, the modified arrayAPI is still an issue which you might have to do a force push (`git push --force`) with a pure ivy repo. (although force pushing is not recommended). Furthermore, please remove your comments in the code. Thanks a lot!

edit: i am not sure if git reset would help here as the arrayAPI is a git submodule. you might have to download/clone the new clean ivy repo and replace the arrayAPI folder manually. Then do a force push.

**Comment by Purity-E** [comment]
I cloned a new repo and made the changes then did a git --force push just
as you had instructed.

**Comment by jieunboy0516** [comment]
hi @Purity-E, I assume you had force pushed into the wrong branch as the arrayAPI change is still there.  The array api change occurs at this [commit](https://github.com/Purity-E/ivy/commit/803d358de0a408aa49301c898b5ecb63e54c42d2) and please revert/force push to change that. Otherwise I am not allowed to merge your PR 😅 Sorry for the inconvenience

**Comment by Purity-E** [comment]
Okay thanks for the clarification. Also I just realized that I didn't do
the manual changes on the array api. I'll do that and then do the git force
push again.

**Comment by jieunboy0516** [comment]
@Purity-E Thanks for your change. Please use `https://unify.ai/docs/ivy/docs/functional/ivy/ivy.functional.ivy.general.html#ivy.get_num_dims` as `AttributeError: 'tensorflow.python.framework.ops.EagerTensor' object has no attribute 'dim'` . The tests are not being passed for NP Jax and TF. Please resolve the errors and make sure all tests are being passed. Thanks

**Comment by jieunboy0516** [comment]
@Purity-E the tests are failing as you are passing in float32 while the function accepts tensor.

**Comment by Purity-E** [comment]
> @Purity-E the tests are failing as you are passing in float32 while the function accepts tensor. please take a look of other unit tests to see how to test tensors. Thanks

@jieunboy0516 this function dtype_and_values(
        available_dtypes=helpers.get_dtypes("float32"),
    ) gives an array and it's dtype,

**Comment by jieunboy0516** [comment]
@Purity-E sorry, i think you are right and you are passing tensors indeed. Could you change the `float32` to `float` as i think `float32` is not supported here somehow and broke the unit tests. Thanks and sorry for the long run! After the quick fix it should be ready for merging.

**Comment by jieunboy0516** [comment]
> > @Purity-E sorry, i think you are right and you are passing tensors indeed. Could you change the `float32` to `float` as i think `float32` is not supported here somehow and broke the unit tests. Thanks and sorry for the long run! After the quick fix it should be ready for merging.
> 
> Thanks. Though I had float initially but it was failing and giving an error about bfloat, I didn't understand the error or how to fix it so I trying it out with float32.

no worries, i know its very confusing

**Comment by Purity-E** [comment]
> > > @Purity-E sorry, i think you are right and you are passing tensors indeed. Could you change the `float32` to `float` as i think `float32` is not supported here somehow and broke the unit tests. Thanks and sorry for the long run! After the quick fix it should be ready for merging.
> > 
> > 
> > Thanks. Though I had float initially but it was failing and giving an error about bfloat, I didn't understand the error or how to fix it so I trying it out with float32.
> 
> no worries, i know its very confusing

Okay

**Pull request merged**